core/src/display_object/edit_text.rs

```diff
@@ -3129,6 +3144,11 @@ impl<'gc> TInteractiveObject<'gc> for EditText<'gc> {
             self.set_selection(None, context.gc());
         }
 
+        if !focused {
+            // Commit IME on focus lost.
+            self.ensure_ime_committed(context);
```

Comment:
Immediately before this is the code:

```rs
self.set_selection(None, context.gc());
```

But `ensure_ime_finished`, which `ensure_ime_committed` calls, has this code:

```rs
self.set_selection(Some(TextSelection::for_position(ime_start)), context.gc());
```

Which selection should happen last in the event of focus being lost?

Reply:
Added a test for this behavior and changed the order of operations. Thanks!